### DISCUSSION TITLE
Interface setting: custom CSS class

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#18013] [Feature Requests]
### Summary

Sometimes we'd like to style single fields different based on the context (e.g highlight important, but not required fields). This also applies to field-groups: It can make sense to give single ones e.g a custom bg-color.

Therefore it would be great if either there's a style setting (specific) or more generic: a custom class setting that can be added to an interface. This would allow us to write custom css to style single fields differently.

I now that e.g input-interfaces do get the field-name as attribute and can therewith be selected. But for example a m2o currently has no chance to be targeted via css.

### Motivation

Add more design-customisation options

### Detailed Design

Add an "custom class" setting to each interface and display. Use the entered class on the interface / display root.

### Must Have:
- Setting
- Class-rendering

### Drawbacks

- More work 😁

### Alternatives

a field-level design setting (overengineered)

### Adoption Strategy

/

**Comment by connorwinston** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
